test/integration/targets/azure_rm_webapp/tasks/main.yml

```diff
@@ -0,0 +1,171 @@
+- name: Fix resource prefix
+  set_fact:
+    linux_app_plan_resource_group: "{{ (resource_prefix | replace('-','x'))[-10:] }}linuxappplan"
+    win_app_name: "{{ (resource_prefix | replace('-','x'))[-8:] }}winapp"
+    win_plan_name: "{{ (resource_prefix | replace('-','x'))[-8:] }}winplan"
+    linux_plan_name: "{{ (resource_prefix | replace('-','x'))[-8:] }}lnxplan"
+
+- name: 1 - Create a windows web app with non-exist app service plan
+  azure_rm_webapp:
+    resource_group: "{{ resource_group }}"
+    name: "{{ win_app_name }}1"
+    plan:
+      resource_group: "{{ resource_group }}"
+      name: "{{ win_plan_name }}"
+      is_linux: false
+      sku: S1
+
+- name: 1 - Create a windows web app with existing app service plan, try to update some root level params
+  azure_rm_webapp:
+    resource_group: "{{ resource_group }}"
+    name: "{{ win_app_name }}2"
+    plan:
+      name: "{{ win_plan_name }}"
+    skip_dns_registration: true
+    https_only: true
+  register: output
+
+- name: create a win web app with java run time specific
+  azure_rm_webapp:
+    resource_group: "{{ resource_group }}"
+    name: "{{ win_app_name }}3"
+    plan:
+      name: "{{ win_plan_name }}"
+    java_version: "1.8"
+    java_container_settings:
+      name: "Tomcat"
+      version: "8.0"
+    app_settings:
+      testkey: "testvalue"
+  register: output
+
+- name: assert the function was created
+  assert:
+    that: output.changed
+
```

Comment:
maybe we should have one idempotence test to check if "not output.changed" if called with the same parameters